framework/pod.go

```diff
@@ -199,23 +199,35 @@ func (f *Framework) DeletePodUntilFinish(name, namespace string, ctx context.Con
 }
 
 func (f *Framework) CheckPodListIpReady(podList *corev1.PodList) error {
+	var v4IpList = make(map[string]string)
+	var v6IpList = make(map[string]string)
 
-	for i := 0; i < len(podList.Items); i++ {
-		if podList.Items[i].Status.PodIPs == nil {
-			return fmt.Errorf("pod %v failed to assign ip", podList.Items[i].Name)
+	for _, pod := range podList.Items {
+		if pod.Status.PodIPs == nil {
+			return fmt.Errorf("pod %v failed to assign ip", pod.Name)
 		}
-		f.Log("pod %v ips: %+v \n", podList.Items[i].Name, podList.Items[i].Status.PodIPs)
+		f.Log("pod %v ips: %+v \n", pod.Name, pod.Status.PodIPs)
 		if f.Info.IpV4Enabled {
-			if !tools.CheckPodIpv4IPReady(&podList.Items[i]) {
-				return fmt.Errorf("pod %v failed to get ipv4 ip", podList.Items[i].Name)
+			ip, ok := tools.CheckPodIpv4IPReady(&pod)
+			if !ok {
+				return fmt.Errorf("pod %v failed to get ipv4 ip", pod.Name)
+			}
+			if _, ok := v4IpList[ip]; ok {
+				return fmt.Errorf("pod %v ipv4 ip %v already exists", pod.Name, ip)
```

Comment:
if d , ok := v4IpList[ip]; ok {
 				return fmt.Errorf("pod %v and %v have conflicted ip %v", pod.Name,  d )
}